### PR TITLE
[SPEC-8103] Update GameLift server instructions for new exposed API

### PR DESCRIPTION
**Pull request by onecent1101**
Signed-off-by: Liu <liug@amazon.com>

### CODE REVIEW

File: content/docs/user-guide/gems/reference/aws/aws-gamelift/cpp-api.md [hunk 1]
```diff
@@ -61,13 +61,13 @@ Searches and retrieves all active sessions that match the provided search criter
...
 ```cpp
-// Make synchronous call
+// For example, make synchronous call to search active game sessions on specific fleet
 AWSGameLift::AWSGameLiftSearchSessionsRequest request;
 request.m_fleetId = "YourGameLiftFleetId";
 AzFramework::SearchSessionsResponse result;
 AWSGameLift::AWSGameLiftSessionRequestBus::BroadcastResult(result, &AWSGameLift::AWSGameLiftSessionRequestBus::Events::SearchSessions, request);
 
-// Make asynchronous call
+// For example, make asynchronous call to search active game sessions on specific fleet and get response from notification
```
Comment (junbo75) [inline content/docs/user-guide/gems/reference/aws/aws-gamelift/cpp-api.md:70]: nit: "on a specific fleet" ?

File: content/docs/user-guide/gems/reference/aws/aws-gamelift/cpp-api.md [hunk 2]
```diff
@@ -85,14 +85,14 @@ Reserves an open player slot in the game session, and initializes a connection f
 To begin the process that allows a player to join the game, call `AWSGameLiftClientManager::JoinSession()` or `AWSGameLiftClientManager::JoinSessionAsync()`, and pass in the game session id and the player id that will join. The process returns `true` if both steps, reserving player slot and initializing connection, succeed. If either step fails, the process returns `false`.
 
 ```cpp
-// Make synchronous call
+// For example, make synchronous call to join specific session
```
Comment (junbo75) [inline content/docs/user-guide/gems/reference/aws/aws-gamelift/cpp-api.md:88]: Nit: "join a specific session" ？

File: content/docs/user-guide/gems/reference/aws/aws-gamelift/cpp-api.md [hunk 3]
```diff
@@ -85,14 +85,14 @@ Reserves an open player slot in the game session, and initializes a connection f
...
 AWSGameLift::AWSGameLiftJoinSessionRequest request;
 request.m_sessionId = "YourGameSessionId";
 request.m_playerId= "YourPlayerId";
 bool result = false;
 AWSGameLift::AWSGameLiftSessionRequestBus::BroadcastResult(result, &AWSGameLift::AWSGameLiftSessionRequestBus::Events::JoinSession, request);
 
-// Make asynchronous call
+// For example, make asynchronous call to join specific session and get response from notification
```
Comment (junbo75) [inline content/docs/user-guide/gems/reference/aws/aws-gamelift/cpp-api.md:95]: Nit: "join a specific session" ？

**Review by junbo75 — COMMENTED** [review]
General question: Did you mention about starting the GameLiftLocal before the launcher if customers want to test it locally in this document?

File: content/docs/user-guide/gems/reference/aws/aws-gamelift/cpp-api.md [hunk 4]
```diff
@@ -112,10 +111,10 @@ Disconnects the player from the game session.
 To leave the game session, call `AWSGameLiftClientManager::LeaveSession()` or `AWSGameLiftClientManager::LeaveSessionAsync()`.
 
 ```cpp
-// Make synchronous call
+// For example, make synchronous call to leave current session
```
Comment (lmbr-pip) [inline content/docs/user-guide/gems/reference/aws/aws-gamelift/cpp-api.md:114]: nit: "leave the current session"

File: content/docs/user-guide/gems/reference/aws/aws-gamelift/cpp-api.md [hunk 5]
```diff
@@ -112,10 +111,10 @@ Disconnects the player from the game session.
...
 AWSGameLift::AWSGameLiftSessionRequestBus::Broadcast(&AWSGameLift::AWSGameLiftSessionRequestBus::Events::LeaveSession);
 
-// Make asynchronous call
+// For example, make asynchronous call to leave current session and get notification after process is complete
```
Comment (lmbr-pip) [inline content/docs/user-guide/gems/reference/aws/aws-gamelift/cpp-api.md:117]: nit: leave the current session and get a notification after the process is complete. 

I would also be wary of "process" here as GameLift has definition of process. Maybe "and get a notification, once the leaving the session is completed"?

File: content/docs/user-guide/gems/reference/aws/aws-gamelift/cpp-api.md [hunk 6]
```diff
@@ -130,22 +129,27 @@ As the default behavior, when the last player leaves the game session, GameLift
 
 ## Server notifications
 
-The AWS GameLift Gem notifies the GameLift service that your server process is ready to host game sessions, handle requests, and take connections.
+You must notify Amazon GameLift service that your server process is ready to host game sessions, handle requests, and take connections.
+
+To send a notification that your server process is ready, use `AWSGameLiftServerRequestBus::Events::NotifyGameLiftProcessReady()`.
+Put this call in a proper place on your server side, like server system component activate step, and call it after you connect your project to `AzFramework::SessionNotificationBus`.
```
Comment (lmbr-pip) [inline content/docs/user-guide/gems/reference/aws/aws-gamelift/cpp-api.md:135]: nit: "proper place" seems to indicate there is a definite place. Maybe "Call process ready once your have completed any relevant initialization. One recommended sport is in the [ServerSystemCompontent]'s activate step, after connecting to the `AzFramework::SessionNotificationBus`

File: content/docs/user-guide/gems/reference/aws/aws-gamelift/cpp-api.md [hunk 7]
```diff
@@ -130,22 +129,27 @@ As the default behavior, when the last player leaves the game session, GameLift
...
+
+```cpp
+AWSGameLift::AWSGameLiftServerRequestBus::Broadcast(&AWSGameLift::AWSGameLiftServerRequestBus::Events::NotifyGameLiftProcessReady);
+```
 
-When your session is active, it broadcasts the following notifications to the GameLift server. You can program how your session responds to these notifications.  
+Once your session is created, it broadcasts the following notifications through `AzFramework::SessionNotificationBus` on server side. You can program how your session responds to these notifications.
```
Comment (lmbr-pip) [inline content/docs/user-guide/gems/reference/aws/aws-gamelift/cpp-api.md:141]: nit: "Once the game session is created, the a notification is broadcast through... You can then listen to the notification to {insert example of what customer should do here}"

File: content/docs/user-guide/gems/reference/aws/aws-gamelift/gem-setup.md [hunk 8]
```diff
@@ -58,11 +58,20 @@ You must include the AWS GameLift Gem static library in your project's CMake bui
                 ...
                 Gem::AWSGameLift.Server.Static
     )
-
     ```
+   
+2. **(Required)** In order to make sure Amazon GameLift Server SDK get initiated correctly before your project server starts, you must mark `AWSGameLiftServerService` as required for your project server system component.
```
Comment (lmbr-pip) [inline content/docs/user-guide/gems/reference/aws/aws-gamelift/gem-setup.md:63]: "To ensure that the Amazon GameLift Server SDK gets intialized correctly...."

**Review by lmbr-pip — COMMENTED** [review]
LG overall.

**Comment by onecent1101** [comment]
> General question: Did you mention about starting the GameLiftLocal before the launcher if customers want to test it locally in this document?

It is in local testing section https://docs.o3de.org/docs/user-guide/gems/reference/aws/aws-gamelift/local-testing/

**Review by junbo75 — APPROVED** [review]

**Review by lmbr-pip — APPROVED** [review]
LG

File: content/docs/user-guide/gems/reference/aws/aws-gamelift/cpp-api.md [hunk 9]
```diff
@@ -130,22 +129,27 @@ As the default behavior, when the last player leaves the game session, GameLift
 
 ## Server notifications
 
-The AWS GameLift Gem notifies the GameLift service that your server process is ready to host game sessions, handle requests, and take connections.
+You must notify Amazon GameLift service that your server process is ready to host game sessions, handle requests, and take connections.
+
+To send a notification that your server process is ready, use `AWSGameLiftServerRequestBus::Events::NotifyGameLiftProcessReady()`.
+Call process ready once your have completed any relevant initialization. One recommended place is in `YourProjectServerSystemComponent` activate step, after connecting to `AzFramework::SessionNotificationBus`.
```
Comment (willihay) [inline content/docs/user-guide/gems/reference/aws/aws-gamelift/cpp-api.md:135]: Update to improve flow and use active voice:
```suggestion
To send a notification that your server process is ready, complete any relevant initialization and then use `AWSGameLiftServerRequestBus::Events::NotifyGameLiftProcessReady()`. We recommend placing the call after connecting to `AzFramework::SessionNotificationBus`, in the `YourProjectServerSystemComponent` activation step.

Example call:
```

File: content/docs/user-guide/gems/reference/aws/aws-gamelift/cpp-api.md [hunk 10]
```diff
@@ -130,22 +129,27 @@ As the default behavior, when the last player leaves the game session, GameLift
...
+
+```cpp
+AWSGameLift::AWSGameLiftServerRequestBus::Broadcast(&AWSGameLift::AWSGameLiftServerRequestBus::Events::NotifyGameLiftProcessReady);
+```
 
-When your session is active, it broadcasts the following notifications to the GameLift server. You can program how your session responds to these notifications.  
+Once the game session is created, following notifications are broadcast through `AzFramework::SessionNotificationBus`. You can program how your session responds to these notifications.
```
Comment (willihay) [inline content/docs/user-guide/gems/reference/aws/aws-gamelift/cpp-api.md:141]:
```suggestion
After the game session has been created, notifications are broadcast through `AzFramework::SessionNotificationBus`. You can program how your session responds to these notifications.
```

File: content/docs/user-guide/gems/reference/aws/aws-gamelift/gem-setup.md [hunk 11]
```diff
@@ -58,11 +58,20 @@ You must include the AWS GameLift Gem static library in your project's CMake bui
                 ...
                 Gem::AWSGameLift.Server.Static
     )
-
     ```
+   
+2. **(Required)** To ensure that the Amazon GameLift Server SDK gets initialized correctly, you must mark `AWSGameLiftServerService` as required for your project server system component.
```
Comment (willihay) [inline content/docs/user-guide/gems/reference/aws/aws-gamelift/gem-setup.md:63]:
```suggestion
2. **(Required)** To ensure that the Amazon GameLift Server SDK gets initialized correctly, you must indicate that the `AWSGameLiftServerService` is required for your project server system component.
```

**Review by willihay — CHANGES_REQUESTED** [review]

File: content/docs/user-guide/gems/reference/aws/aws-gamelift/cpp-api.md [hunk 12]
```diff
@@ -130,22 +129,27 @@ As the default behavior, when the last player leaves the game session, GameLift
 
 ## Server notifications
 
-The AWS GameLift Gem notifies the GameLift service that your server process is ready to host game sessions, handle requests, and take connections.
+You must notify Amazon GameLift service that your server process is ready to host game sessions, handle requests, and take connections.
+
+To send a notification that your server process is ready, complete any relevant initialization and then use `AWSGameLiftServerRequestBus::Events::NotifyGameLiftProcessReady()`.
+We recommended placing the call after connecting to `AzFramework::SessionNotificationBus` in the `YourProjectServerSystemComponent` activate step.
```
Comment (willihay) [inline content/docs/user-guide/gems/reference/aws/aws-gamelift/cpp-api.md:135]:
```suggestion
We recommend placing the call after connecting to `AzFramework::SessionNotificationBus` in the `YourProjectServerSystemComponent` activate step.
```

**Review by willihay — CHANGES_REQUESTED** [review]

**Review by willihay — APPROVED** [review]

**Pull request merged**